tygron-environment/src/main/java/nl/tudelft/contextproject/tygron/SessionManager.java

```diff
@@ -51,14 +50,16 @@ public SessionManager(Connection localApiConnection){
    * @return whether the join was successful or not.
    */
   public boolean joinSession(Session session, int slotId) {
```

Comment:
The problem here is that with joining you do not know the slotid 'yet'. So it's either giving it here, or create a session somewhere, set the id manually and then join after all.